pulp_puppet_plugins/pulp_puppet/plugins/migrations/0005_puppet_module_name_change.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+from mongoengine import NotUniqueError
+
+from pulp.server.controllers import repository as repo_controller
+from pulp.server.db import model
+
+from pulp_puppet.plugins.db.models import Module
+
+_log = logging.getLogger('pulp')
+
+
+def migrate(*args, **kwargs):
+    """
+    For each puppet module check and if needed update module name format.
+    """
+    modules = Module.objects(__raw__={'name': {'$regex': '/|-'}})
+    repos_to_rebuild = set()
+    for puppet_unit in modules:
+        try:
+            author, name = puppet_unit['name'].split("-", 1)
+        except ValueError:
+            # This is the forge format, but Puppet still allows it
+            author, name = puppet_unit['name'].split("/", 1)
+        try:
+            puppet_unit.update(set__name=name)
+        except NotUniqueError:
+            # find all repos that have this unit
+            repos_with_unit = model.RepositoryContentUnit.objects(unit_id=puppet_unit.id)
+            repos_to_rebuild.update(repos_with_unit)
+            # find unit with correct name
+            correct_unit = Module.objects(__raw__={'name': name})
```

Comment:
Similar querying change here, replacing the `__raw__` with a call to `Module.objects.filter(...)`